src/Stryker.Core/Stryker.Core/Initialisation/InputFileResolver.cs

```diff
@@ -152,10 +153,12 @@ private void MarkInputFilesAsExcluded(FolderComposite root, List<string> pathsTo
             }
         }
 
-        public string ScanProjectFile(string currentDirectory)
+        public string FindProjectFile(string currentDirectory, string testProjectNameFilter)
         {
-            var projectFiles = _fileSystem.Directory.GetFiles(currentDirectory, "*.csproj");
-            _logger.LogTrace("Scanned the current directory for *.csproj files: found {0}", projectFiles);
+            var projectFiles = _fileSystem.Directory.GetFiles(currentDirectory, testProjectNameFilter);
```

Comment:
What will happen when the path `../../myproject.csproj` is passed as filter?Is this a valid filter for Directory.GetFiles? Or the path `C:/dev/myproject/mytestproject/mytestproject.csproj`? 

Also, do we validate what kind of filter is passed? I think we can already rule out wrong filters like a missing `csproj` in the end.

Reply:
A filter without csproj would still work as long as the partial file name matches only one file

Reply:
The docs say the following:

```
searchPattern cannot end in two periods ("..") or contain two periods ("..") followed by DirectorySeparatorChar or AltDirectorySeparatorChar, nor can it contain any invalid characters. You can query for invalid characters by using the GetInvalidPathChars method.
```

So it seems like relative paths are not supported

Reply:
@richardwerkman I fixed the relative and full paths by converting relative paths with navigation characters to full paths in StrykerOptions so the filter always contains either a full path or a relative path without navigation characters. In the FindProjectFile method I remove the basePath from the full path if it exists, so it will at that point be converted to a relative path without navigation. This should fix all cases I think.

Reply:
Cool! Looks good :+1: